### PR TITLE
Bug1984431 non ca hsm 2 step install v10.11

### PR DESCRIPTION
**Pull request by ladycfu**

### CODE REVIEW

File: base/common/python/pki/nssdb.py [hunk 1]
```diff
@@ -815,6 +817,54 @@ def create_request(
...
+            token = self.get_effective_token(token)
+            password_file = self.get_password_file(tmpdir, token)
+
+            if token:
+                cmd.extend(['-h', token])
+
+            if password_file:
+                with open(password_file, 'r') as f:
+                    password = f.read()
+                cmd.extend(['-p', password])
```
Comment (edewata) [inline base/common/python/pki/nssdb.py:850]: Does `PKCS10Client` have any option to specify a password file? Usually it's not recommended to specify a password as a command line parameter since it could expose the password in the logs.

File: base/common/python/pki/nssdb.py [hunk 2]
```diff
@@ -1196,7 +1252,14 @@ def get_trust(self, nickname, token=None):
...
             re_compile = re.compile(r'^' + fullname + r'\s+(\S+)$', re.MULTILINE)
-            cert_trust = re.search(re_compile, output.decode()).group(1)
+            re_line = re.search(re_compile, output.decode())
+            if re_line != None:
+                print(re_line)
+                cert_trust = re.search(re_compile, output.decode()).group(1)
+                logger.debug('cert_trust : %s', cert_trust)
+            else:
+                logger.debug('re_line for re.search is None')
+                print("output.decode : %s", output.decode())
```
Comment (edewata) [inline base/common/python/pki/nssdb.py:1262]: Are these changes necessary? It seems to be executing the same regular expression twice.

File: base/server/python/pki/server/deployment/scriptlets/keygen.py [hunk 3]
```diff
@@ -93,23 +93,44 @@ def generate_csr(self,
...
+option of PKCS10Client
+e.g. PKCS10Client -d /var/lib/pki/<ca instance>/alias -h hsm-module -a rsa -l 2048 -n "CN=CA issuanceProt cert" -w -v -o ca-issuanceProt-cfu.csr.b64
+
+        """
+
+        print('pki_hsm_enable: ', config.str2bool(deployer.mdict['pki_hsm_enable']))
+        print('subsystem.type: ', subsystem.type)
+        if (subsystem.type == 'KRA' and 
+            config.str2bool(deployer.mdict['pki_hsm_enable']) and
+                (cert_id in ['storage', 'transport'])):
```
Comment (edewata) [inline base/server/python/pki/server/deployment/scriptlets/keygen.py:108]: Should it check to make sure that `key_type` is `rsa`? If it's not `rsa`, should it proceed to create an RSA key anyway or should it fail?

File: base/server/python/pki/server/subsystem.py [hunk 4]
```diff
@@ -224,8 +227,8 @@ def validate_system_cert(self, cert_id=None):
             print(subprocess.check_output(
                 cmd,
                 stderr=subprocess.STDOUT))
-        except subprocess.CalledProcessError as e:
-            print("pki-server subsystem-cert-validate stdout output:\n", e.output)
+        except subprocess.CalledProcessError as v:
+            print("pki-server subsystem-cert-validate stdout output:\n", v.output)
```
Comment (edewata) [inline base/server/python/pki/server/subsystem.py:231]: Sorry I missed your question. The original code combines the stdout and stderr. If you want to separate them out you probably could do this (I haven't tried it myself):
```
result = subprocess.run(cmd, capture_output=True, text=True)
print(result.stdout.strip('\n'))
if result.returncode == 0:
    print(result.stderr.strip('\n'))
```
https://stackoverflow.com/questions/41171791/how-to-suppress-or-capture-the-output-of-subprocess-run

**Review by edewata — APPROVED** [review]
I just have some minor comments, but it LGTM. Feel free to update if you wish and merge.

File: base/server/python/pki/server/deployment/scriptlets/keygen.py [hunk 5]
```diff
@@ -93,23 +96,54 @@ def generate_csr(self,
...
+        print('pki_hsm_enable: ', config.str2bool(deployer.mdict['pki_hsm_enable']))
+        print('pki_use_pss_rsa_signing_algorithm: ',
+              config.str2bool(deployer.mdict['pki_use_pss_rsa_signing_algorithm']))
+
+        print('subsystem.type: ', subsystem.type)
+        if (subsystem.type == 'KRA' and
+            config.str2bool(deployer.mdict['pki_hsm_enable']) and
+            config.str2bool(deployer.mdict['pki_use_pss_rsa_signing_algorithm']) and
+                (cert_id in ['storage', 'transport'])):
+            print('generate_csr: calling PKCS10Client for', cert_id)
```
[the remark this answers is not in the document]
Comment (ladycfu) [reply base/server/python/pki/server/deployment/scriptlets/keygen.py:122]: actually, for some reason, it got me a space... here is my screen capture:
generate_csr: calling PKCS10Client for storage

[the remark this answers is not in the document]
Comment (edewata) [reply base/server/python/pki/server/deployment/scriptlets/keygen.py:122]: I guess `print()` will insert a space automatically, so never mind then:
https://stackoverflow.com/questions/15286401/print-multiple-arguments-in-python
